### PR TITLE
Fix depth <-> color copies

### PR DESCRIPTION
**Pull request by etang-cw**
Metal doesn't let you make color views of depth textures

### CODE REVIEW

File: MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm [hunk 1]
```diff
@@ -133,12 +133,14 @@ static inline MTLSize mvkClampMTLSize(MTLSize size, MTLOrigin origin, MTLSize ma
         
         MTLPixelFormat srcMTLPixFmt = _srcImage->getMTLPixelFormat(srcPlaneIndex);
         bool isSrcCompressed = _srcImage->getIsCompressed();
+        bool isSrcDepth = _srcImage->getIsDepthStencil();
 
         MTLPixelFormat dstMTLPixFmt = _dstImage->getMTLPixelFormat(dstPlaneIndex);
         bool isDstCompressed = _dstImage->getIsCompressed();
+        bool isDstDepth = _dstImage->getIsDepthStencil();
 
         // If source and destination have different formats and at least one is compressed, use a temporary intermediary buffer
```
Comment (cdavis5e) [inline MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm:142]: You should probably update this comment, too.

**Review by cdavis5e — APPROVED** [review]
Pretty straightforward. Just one minor quibble...

File: MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm (hunk 1 above)
Comment (billhollings) [inline MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm:142]:
```suggestion
        // If source and destination have different formats and at least one is compressed, or one is color and the other is depth/stencil, use a temporary intermediary buffer.
```

**Review by billhollings — CHANGES_REQUESTED** [review]
Thanks for the submission!

Per the observation by @cdavis5e, please update the comment (for which I've made a suggestion, so you can accept it here).

File: MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm (hunk 1 above)
Comment (billhollings) [reply MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm:142]: Meh. I'll just do it here. 😉

**Review by billhollings — APPROVED** [review]

**Comment by etang-cw** [comment]
Actually, it looks like depth <-> color copies aren't allowed in Vulkan, and I should be fixing vkd3d

Do we still want this?

**Comment by cdavis5e** [comment]
> Do we still want this?

In that case, it's probably better to close this, since this is undefined behavior anyway. I can't speak for @billhollings, but in cases of undefined behavior, I prefer that the driver do the least amount of work necessary.

**Comment by billhollings** [comment]
> in cases of undefined behavior, I prefer that the driver do the least amount of work necessary

Agreed. If this is forbidden in Vulkan too, then this should be triggering a Validation Layer error, and if not, should be fixed there.

**Pull request closed without merging**